src/services/task.service.ts

```diff
@@ -0,0 +1,57 @@
+import { DateTime } from 'luxon';
+
+import { taskSchema } from '../models/task.model';
+import { TaskCreatePayload } from '../controllers/v1/task/create.controller';
+import { CreateTaskDto, TaskRepository } from '../repositories/task/create';
+import { TaskStartFinishTimeError, TaskTimelineError } from '../errors/error';
+
+export class TaskService {
+  async createTask(data: TaskCreatePayload) {
+    const taskRepo = new TaskRepository(taskSchema);
+
+    const { startTime, completionTime } = data;
+    this.validateTimeline({ startTime, completionTime });
+
+    const newTask = await taskRepo.createTask(this.mapTaskObject(data));
+
+    if (newTask.kind === 'ok') {
+      return newTask.data;
+    }
+
+    throw newTask.err;
+  }
+
+  private validateTimeline(timelime: { startTime?: DateTime; completionTime?: DateTime }) {
+    const { startTime, completionTime } = timelime;
+
+    const curTime = DateTime.utc();
+    if (startTime && startTime < curTime) {
+      throw new TaskStartFinishTimeError('startTime');
+    }
+    if (completionTime && completionTime < curTime) {
+      throw new TaskStartFinishTimeError('completionTime');
+    }
+    if (startTime && completionTime) {
+      if (completionTime <= startTime) {
+        throw new TaskTimelineError('');
+      }
+    }
+  }
+
+  private mapTaskObject(d: TaskCreatePayload) {
+    const mapped = Object.entries(d).reduce(
+      (acc, [key, val]) => {
+        if (val instanceof DateTime) {
+          acc[key] = val.toJSDate();
+        } else {
+          acc[key] = val;
+        }
+
+        return acc;
+      },
+      {} as { [key: string]: any }
+    );
+
+    return mapped as CreateTaskDto;
```

Comment:
Prefer explicit typing to casting using `as`

Reply:
This is an area of concern. Could you suggest how to accomplish this given the use of the index signature `{ [key: string]: any }` present?